// bizgoblin/es/plugin/PieLabel.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import * as React from 'react';
import * as PropTypes from 'prop-types';
import { Chart } from 'goblin-base';
import register from 'goblin-base/es/plugin/pieLabel';
var Props = (function () {
    function Props() {
    }
    return Props;
}());
;
var PieLabel = (function (_super) {
    __extends(PieLabel, _super);
    function PieLabel(props) {
        var _this = _super.call(this, props) || this;
        _this.displayName = 'PieLabel';
        return _this;
    }
    PieLabel.prototype.componentDidMount = function () {
        register(Chart);
        this.context.centralizedUpdates(this);
    };
    PieLabel.prototype.componentDidUpdate = function () {
        this.context.centralizedUpdates(this);
    };
    PieLabel.prototype.render = function () {
        return null;
    };
    PieLabel.contextTypes = {
        centralizedUpdates: PropTypes.func
    };
    return PieLabel;
}(React.Component));
export default PieLabel;
